src/Compiler/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/CodeWriterExtensions.cs

```diff
@@ -564,7 +564,7 @@ private static SourceSpan RemapFilePathIfNecessary(SourceSpan sourceSpan, CodeRe
             // If you try and use the line pragma in the design time docs to map back to the original file it will fail,
             // as the path isn't actually valid on windows. As a workaround we apply a simple heuristic to switch the
             // paths back when writing out the design time paths.
-            sourceSpan = new SourceSpan(sourceSpan.FilePath.Replace("/", "\\"), sourceSpan.AbsoluteIndex, sourceSpan.LineIndex, sourceSpan.CharacterIndex, sourceSpan.Length);
+            sourceSpan = new SourceSpan(sourceSpan.FilePath.Replace("/", "\\"), sourceSpan.AbsoluteIndex, sourceSpan.LineIndex, sourceSpan.CharacterIndex, sourceSpan.Length, sourceSpan.LineCount, sourceSpan.EndCharacterIndex);
```

Comment:
Shouldn't we have a test for this?

Reply:
We do have tests, but they were all at the start of the doc so the baseline didn't actually change. I'll add a test that covers it.

Reply:
Updated test